### PR TITLE
[MIG][11.0] hr_holidays_validity_date

### PR DESCRIPTION
**Pull request by jarroyomorales**
Migration to 11.0 of hr_holidays_validity_date

### CODE REVIEW

**Review by JordiBForgeFlow — APPROVED** [review]

File: hr_holidays_validity_date/models/hr_holidays.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    _inherit = "hr.holidays.status"
+
+    date_start = fields.Datetime(string='Start Date')
+    date_end = fields.Datetime(string="Expiry Date")
+    use_validity_dates = fields.Boolean()
+
+    _sql_constraints = [
+        ('date_check',
+         "CHECK ( (use_validity_dates=FALSE) OR (date_start <= date_end))",
+         "The start date must be anterior to the end date."),
```
Comment (JordiBForgeFlow) [inline hr_holidays_validity_date/models/hr_holidays.py]: Careful with the translation!

**Comment by JordiBForgeFlow** [comment]
@jarroyomorales In order to bring the expected behaviour of just displaying a warning that you implemented in https://github.com/OCA/hr/pull/596, I suggest that in the holiday.status, next to the interval you define a check 'mandatory'. If it's set, then the error message is triggered. Otherwise the warning is shown.

File: hr_holidays_validity_date/models/hr_holidays.py (hunk 1 above)
Comment (jarroyomorales) [reply hr_holidays_validity_date/models/hr_holidays.py]: I have seen this sentence being used by odoo or in other OCA modules but I can change it for "The start date must be greater than the end date."

**Comment by jarroyomorales** [comment]
@jbeficent Okay, great idea, I'm going to implement it. Thank you!

**Comment by jarroyomorales** [comment]
@jbeficent I just pushed it, what do you think?

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: LGTM functional tested. Just a minor comment

**Comment by jarroyomorales** [comment]
@aheficent 

> +1 LGTM functional tested. Just a minor comment

Were you going to say something?

File: hr_holidays_validity_date/models/hr_holidays.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import pytz
+
+
+class HrHolidaysStatus(models.Model):
+    _inherit = "hr.holidays.status"
+
+    date_start = fields.Datetime(string='Start Date')
+    date_end = fields.Datetime(string="Expiry Date")
+    use_validity_dates = fields.Boolean()
+    restrict_dates = fields.Boolean(string="Restrict")
```
Comment (AaronHForgeFlow) [inline hr_holidays_validity_date/models/hr_holidays.py]: IMO better it is better Restrict Dates as string. It is more clear in the view.

Comment (AaronHForgeFlow) [reply hr_holidays_validity_date/models/hr_holidays.py]: Sorry @jarroyomorales. This is the comment. For some reason it did not appear before. Regards.

Comment (jarroyomorales) [reply hr_holidays_validity_date/models/hr_holidays.py]: Changed!

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**